Advanced-Recruit-Training/src/BatteryStateOfChargeService.h

```diff
@@ -25,4 +27,13 @@ class BatteryStateOfChargeService : public I_BatteryStateOfChargeService
     const double initialStateOfChargePercent_;
 
     // Add your necessary variables here
+    double current_;
+    double voltage_;
+    QTime time_;
+    double previousCurrent_;
+    QTime previousTime_;
+    double ampHoursUsed_;
+    double averageCurrent_;
```

Comment:
This doesn't need to be a member variable, it's only used in `addData` and it's value is not persistent between calls.